pandas/core/frame.py

```diff
@@ -3267,9 +3267,12 @@ def memory_usage(self, index: bool = True, deep: bool = False) -> Series:
             index=self.columns,
         )
         if index:
-            result = self._constructor_sliced(
+            from pandas.core.reshape.concat import concat
+
+            index_memory_usage = self._constructor_sliced(
                 self.index.memory_usage(deep=deep), index=["Index"]
-            ).append(result)
+            )
+            result = concat([index_memory_usage, result])
```

Comment:
i'd rather keep using _append internally; should be more performant and avoids circular dependency